app/models/agents/naive_bayes_agent.rb

```diff
@@ -0,0 +1,509 @@
+      #
+      # P(wi|class) = (count(wi, class) + k)/(count(w,class) + kV)
+      prob_numerator = {}
+      v_size = vocab.size
+
+      cat_prob = Math.log(1 / data.categories.count.to_f)
+      total_example_count = data.total_examples.to_f
+
+      data.each do |category|
+        unless assume_uniform
+          cat_prob = Math.log(data.example_count(category) / total_example_count)
+        end
+
+        log_probs = 0
+        denominator = (data.token_count(category) + @k * v_size).to_f
+        tokens.each do |token|
+          numerator = data.count_of_token_in_category(category, token) + @k
+          log_probs += Math.log( numerator / denominator )
+        end
+        prob_numerator[category] = log_probs + cat_prob
+      end
+      normalize(prob_numerator)
+    end
+
+    def normalize(prob_numerator)
+      # calculate the denominator, which normalizes this into a probability; it's just the sum of all numerators from above
+      normalizer = 0
+      prob_numerator.each {|cat, numerator| normalizer += numerator }
+      # One more caveat:
+      # We're using log probabilities, so the numbers are negative and the smallest negative number is actually the largest prob.
+      # To convert, we need to maintain the relative distance between all of the probabilities:
+      # - divide log prob by normalizer: this keeps ratios the same, but reverses the ordering
+      # - re-normalize based off new counts
+      # - final calculation
+      # Ex: -1,-1,-2  =>  -4/-1, -4/-1, -4/-2
+      #   - renormalize and calculate => 4/10, 4/10, 2/10
+      intermed = {}
+      renormalizer = 0
+      prob_numerator.each do |cat, numerator|
+        intermed[cat] = normalizer / numerator.to_f
+        renormalizer += intermed[cat]
+      end
+      # calculate final probs
+      final_probs = {}
+      intermed.each do |cat, value|
+        final_probs[cat] = value / renormalizer.to_f
+      end
+      final_probs
+    end
+
+    # called internally after yaml import to reset Hash defaults
+    def reset_after_import
+      data.reset_after_import
+    end
+
+    def self.from_yml(yml_data)
+      nbayes = YAML.load(yml_data)
+      nbayes.reset_after_import()  # yaml does not properly set the defaults on the Hashes
+      nbayes
+    end
+
+    # Loads class instance from a data file (e.g., yaml)
+    def self.from(yml_file)
+      File.open(yml_file, "rb") do |file|
+        self.from_yml(file.read)
+      end
+    end
+
+    # Load class instance
+    def load(yml)
+      if yml.nil?
+        nbayes = NBayes::Base.new
+      elsif yml[0..2] == "---"
+        nbayes = self.class.from_yml(yml)
+      else
+        nbayes = self.class.from(yml)
+      end
+      nbayes
+    end
+
+    # Dumps class instance to a data file (e.g., yaml) or a string
+    def dump(arg)
+      if arg.instance_of? String
+        File.open(arg, "w") {|f| YAML.dump(self, f) }
+      else
+        YAML.dump(self) # XXX only line modified so far for Huginn Agent
```

Comment:
I just checked and this function can just be avoided without much hassle. Instead of calling `nbays.dump` the code can just say `YAML.dump(nbayes)`. So I'll make this change and then there won't be a need to modify NBayes. 

It might be worth it to still make the pull request just to add the useful functionality to NBayes.